From 553f0cd6090bd44c26cea6e0425c22590bdb41d2 Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Fri, 21 Jul 2017 02:18:02 -0700
Subject: [PATCH 01/22] Fix a logical error in function bool
 evtlog_append(...).

There is a logical error in function
bool evtlog_append(uint8_t pcr, hash_list_t *hl, uint32_t type).
The function always returns true whether event log append is
successful or not for both TPM1.2 or TPM2.0.

Signed-off-by: Shi Wangyi <shiwangyi@gohighsec.com>
Reviewed-by: Ning Sun <ning.sun@intel.com>
(cherry picked from commit 9e6a264ddc64a713c771342f388e5b9dde3d4753)
---
 tboot/txt/txt.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/tboot/txt/txt.c b/tboot/txt/txt.c
index b920b28..44d0d23 100644
--- a/tboot/txt/txt.c
+++ b/tboot/txt/txt.c
@@ -430,12 +430,15 @@ bool evtlog_append(uint8_t pcr, hash_list_t *hl, uint32_t type)
 {
     switch (g_tpm->major) {
     case TPM12_VER_MAJOR:
-        evtlog_append_tpm12(pcr, &hl->entries[0].hash, type);
+        if ( !evtlog_append_tpm12(pcr, &hl->entries[0].hash, type) )
+            return false;
         break;
     case TPM20_VER_MAJOR:
-        for (unsigned int i=0; i<hl->count; i++)
-            evtlog_append_tpm20(pcr, hl->entries[i].alg,
-                    &hl->entries[i].hash, type);
+        for (unsigned int i=0; i<hl->count; i++) {
+            if ( !evtlog_append_tpm20(pcr, hl->entries[i].alg,
+                    &hl->entries[i].hash, type))
+                return false;
+	}
         break;
     default:
         return false;
-- 
2.16.1

